e2e/cli-e2e/tests/history.e2e.test.ts

```diff
@@ -0,0 +1,53 @@
+import { readdir } from 'node:fs/promises';
+import { join } from 'node:path';
+import { PluginReport, Report, reportSchema } from '@code-pushup/models';
+import { cleanTestFolder } from '@code-pushup/test-setup';
+import { executeProcess, readJsonFile } from '@code-pushup/utils';
+
+describe('CLI history', () => {
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  const omitVariableData = ({
+    date,
+    duration,
+    version,
+    ...report
+  }: Omit<Report, 'commit'> | PluginReport) => report;
+  const omitVariableReportData = ({ commit, ...report }: Report) =>
+    omitVariableData({
+      ...report,
+      plugins: report.plugins.map(omitVariableData) as PluginReport[],
+    });
+  /* eslint-enable @typescript-eslint/no-unused-vars */
+
+  beforeEach(async () => {
+    await cleanTestFolder('tmp/e2e');
+  });
+
+  it('should run ESLint plugin and create report.json for the last 5 commits', async () => {
```

Comment:
In our E2E tests, we should not run the commands on our own repository. Since the `history` command also forces a clean state, it would delete people's progress, that is a hard no.

A way of testing a `history` command would be to create a temporary repository, say with the `todos-app` data (so 2 commits, relatively small codebase) temporarily, so that our original repo is not affected and the number of commits is minimal - so 2 (5 is pushing it and could take a long time).

One could then run the report for that temporary repository and delete it at the end of the test.

One important question though is to ask oneself if the time spent on the setup is worth the payoff.

Reply:
I'm happy to skip e2e :)

Reply:
If we were to go down that route, then @Tlacenka's approach is a good one - something along the lines of `const git = simpleGit('examples/react-todos-app'); git.init(); ...` and always clean up afterwards. In the [E2E tests for the `compare` command](https://github.com/code-pushup/cli/blob/main/e2e/cli-e2e/tests/compare.e2e.test.ts), I actually make changes to the example app and clean them up afterwards, that's quite similar. The only extra prerequisite is that there are no unstaged changes to the example app.

On the other hand, I wouldn't yet consider the `history` command as a critical flow. I understand it's for one-off usages by us, not something every customer will run in their CI. So I'm leaning towards E2E tests not being worth the hassle in this case.

Reply:
ok

Reply:
added a issue for the additional tests #568

Reply:
In order for this PR to be merged, please either _remove these E2E tests_, or rewrite them as discussed _in this PR_. I don't want to merge anything which messes around with our actual Git repo, particularly if it's using a force flag to do it.

Reply:
Removed